### ISSUE TITLE
Undo text not working when focus is in file explorer

### ISSUE THREAD

**Issue by aeschli** [microsoft/vscode#111630]
Testing #111015

- In the VSCode repo, open `src/vs/nls.build.js`
- add some content on the first line, save
- in the file explorer rename the file to `nls.build.js1`
- from the window menu, run `Edit > Undo`. File is renamed back :heavy_check_mark: 
- from the window menu, run `Edit > Undo`. Nothing happens, content change is not undone
- but the focus in the editor, then undo works

**Comment by isidorn** [comment]
We have decide to behave like this -> if focus is in explorer we do not undo the changes done in the editor.
We can easily change this, though I would like to gather more feedback before chaning anything here. Thus assigning to December and debt week.

**Comment by alexdima** [comment]
I've changed the title to reflect easier repro steps:
* type in editor
* click in explorer
* try to undo

**Comment by isidorn** [comment]
The only negative feedback regarding undo is this issue https://github.com/microsoft/vscode/issues/112706

Making the undo stack "shared" between the explorer and editor does not make a lot of sense to me since it is not really complementery with the current approach of each editor having it's own undo stack.

The only thing I would consider changing is: if the Explorer has no undo stack, then pressing undo when focus is in the Explorer would undo in the editor. @alexdima what do you think?

**Comment by alexdima** [comment]
> The only negative feedback regarding undo is this issue #112706

@isidorn I agree that we should prompt for destructive undos. Is that something I need to help you with or can you do it outside the undo redo service?

> if the Explorer has no undo stack, then pressing undo when focus is in the Explorer would undo in the editor

I like this idea. I think you can do it on your side by doing something like this e.g.

![image](https://user-images.githubusercontent.com/5047891/104780449-1d5d8d00-5781-11eb-9df8-d5e8af235fd5.png)

**Comment by isidorn** [comment]
@alexdima thanks. I just tried that and works as expected. Thus I pushed and closing this one since I would do no other changes here.

As for the prompt for desctructive undo, I think I can handle it from the `DeleteOperation` that is an `undo`.
I will create a PR to tackle that and we can continue the discussion https://github.com/microsoft/vscode/issues/112706

**Comment by isidorn** [comment]
Adding bug label so we get this verified, steps:

* Reload vscode
* Do some edits in the editor
* Focus the explorer, press cmd + Z, the editor should undo

**Comment by isidorn** [comment]
@alexdima on second thought I just remembered that the `UndoRedoService` already has a check [here](https://github.com/microsoft/vscode/blob/667e41626f111ef9e77e10542e689798484b5013/src/vs/platform/undoRedo/common/undoRedoService.ts#L941)

So if I implement this in the `DeleteOperation` I might be duplicating code.
However I see that the `UndoRedoService` is unaware of what kind of an operation it is.

To sum it up I suggest:
* Let's remove the check from the `UndoRedoService`
* And have only a check in the `DeleteOperation`. That should catch both the new case, and the original case for which you introduced the check in the `UndoRedoService`

Let me know what you think

fyi @jrieken
